Use compat_semaphore for a real semaphore

On preempt-RT the 'struct semaphore' is a mutex.
This is because originally the semaphore was used as a mutex in
a lot of places in the kernel. Only on a limited number of places
the semaphore was used as a normal counting semaphore. These
low amount of instances have been changed to 'compat_semaphore'

Signed-off-by: Remy Bohmer <linux@bohmer.net> 
---
 drivers/video/mxc/mxc_ipuv3_fb.c |    2 +-
 drivers/video/mxc/mxcfb.c        |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.24/drivers/video/mxc/mxc_ipuv3_fb.c
===================================================================
--- linux-2.6.24.orig/drivers/video/mxc/mxc_ipuv3_fb.c	2009-01-26 20:34:24.000000000 +0100
+++ linux-2.6.24/drivers/video/mxc/mxc_ipuv3_fb.c	2009-01-26 20:35:00.000000000 +0100
@@ -65,7 +65,7 @@ struct mxcfb_info {
 
 	u32 pseudo_palette[16];
 
-	struct semaphore flip_sem;
+	struct compat_semaphore flip_sem;
 	struct completion vsync_complete;
 };
 
Index: linux-2.6.24/drivers/video/mxc/mxcfb.c
===================================================================
--- linux-2.6.24.orig/drivers/video/mxc/mxcfb.c	2009-01-26 20:34:24.000000000 +0100
+++ linux-2.6.24/drivers/video/mxc/mxcfb.c	2009-01-26 20:35:28.000000000 +0100
@@ -63,7 +63,7 @@ struct mxcfb_info {
 
 	u32 pseudo_palette[16];
 
-	struct semaphore flip_sem;
+	struct compat_semaphore flip_sem;
 	spinlock_t fb_lock;
 };
 
